actor_library/src/access_control/access_control.rs

```diff
@@ -0,0 +1,236 @@
+use cid::Cid;
+use fil_actors_runtime::{ActorError, AsActorError};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_hamt::BytesKey;
+use fvm_shared::{
+    address::Address,
+    error::ExitCode,
+    event::{Entry, Flags},
+    IPLD_RAW,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::context::Context;
+use crate::utils::map;
+
+const ROLE_ADMIN_CHANGED_EVENT_KEY: &str = "role_admin_changed_event";
+const ROLE_GRANTED_EVENT_KEY: &str = "role_granted_event";
+const ROLE_REVOKED_EVENT_KEY: &str = "role_revoked_event";
+
+#[derive(Serialize, Deserialize)]
+pub struct RoleAdminChangedEvent {
+    pub role: Role,
+    pub previous_admin_role: Role,
+    pub new_admin_role: Role,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RoleGrantedEvent {
+    pub role: Role,
+    pub account: Address,
+    pub sender: Address,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct RoleRevokedEvent {
+    pub role: Role,
+    pub account: Address,
+    pub sender: Address,
+}
+
+pub type Role = u32;
+
+pub trait AccessControl {
+    // Modifier
+    fn only_role<BS: Blockstore>(&self, store: &BS, role: Role);
+
+    fn has_role<BS: Blockstore>(&self, store: &BS, role: Role, account: Address) -> bool;
+
+    fn get_role_admin<BS: Blockstore>(&self, store: &BS, role: Role) -> Role;
+
+    fn grant_role<BS: Blockstore>(&mut self, store: &BS, role: Role, account: Address);
```

Comment:
Methods that can throw an error should return anyhow::Result<_>

```suggestion
    fn grant_role<BS: Blockstore>(&mut self, store: &BS, role: Role, account: Address) -> anyhow::Result<()>;
```

Reply:
Fixed [here](https://github.com/Ethernal-Tech/filecoin-actor-library/pull/5/commits/dcd00ad0e72ae3cb2412881b9b4a8a415dcf9860#diff-324b0e57c0074407a39059591d361b952bc9f0f06b445d4376767d222ba55ef0R239)